calico_containers/libnetwork_plugin/datastore_libnetwork.py

```diff
@@ -0,0 +1,35 @@
+from etcd import EtcdKeyNotFound
+import json
+from pycalico.ipam import IPAMClient
+
+PREFIX = "/CALICO_DOCKER/"
+#TODO comments docstrings
+
+class LibnetworkDatastoreClient(IPAMClient):
+    def cnm_endpoint_exists(self, ep_id):
+        endpoint_path = PREFIX+ep_id
+        try:
+            _ = self.etcd_client.read(endpoint_path)
+        except EtcdKeyNotFound:
+            return False
+        else:
+            return True
+
+    def read_cnm_endpoint(self, ep_id):
+        try:
+            endpoint = self.etcd_client.read(PREFIX+ep_id)
+            return json.loads(endpoint.value)
+        except EtcdKeyNotFound:
+            return None
+
+
+
+    def write_cnm_endpoint(self, ep_id, ep):
+        self.etcd_client.write(PREFIX+ep_id, json.dumps(ep))
+
+    def delete_cnm_endpoint(self, ep_id):
+        try:
+            self.etcd_client.delete(PREFIX+ep_id)
+            return True
```

Comment:
Not really sure what the standards are here, but I'm usually a fan of putting minimum amount in the actual try:, so I'd remove the return True and put that in an else: clause.